### PR TITLE
[new release] config (0.0.2)

### PR DESCRIPTION
**Pull request by leostera**
Ergonomic, lightweight conditional compilation through attributes

- Project page: <a href="https://github.com/ocaml-sys/config.ml">https://github.com/ocaml-sys/config.ml</a>

## 0.0.2

* Fix single var configuration failing – thanks @metame 👏
* Support conditionally disabling all structure-items
* Support conditionally disabling variant constructors and record fields
* Add better docs for the Config language, and a small cookbook

### CODE REVIEW

**Comment by haochenx** [comment]
There are three CI failures, all during testing, two due to 32-bit incompatibility and one due to OCaml 5.2 detecting `misplaced-attribute`.

compilers
  - 5.2~alpha1
    - config.0.0.2
[tests (failed: The compilation of config.0.0.2 failed at "dune build -p config -j 39 @install @runtest".)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/16922a5c4f8c89c2528a3ae77b454682ec9b814b/variant/compilers,5.2~alpha1,config.0.0.2,tests)

extras
  - arm32-ocaml-5.1
    - config.0.0.2
[tests (failed: The compilation of config.0.0.2 failed at "dune build -p config -j 79 @install @runtest".)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/16922a5c4f8c89c2528a3ae77b454682ec9b814b/variant/extras,arm32-ocaml-5.1,config.0.0.2,tests)
  - x86_32-ocaml-5.1
    - config.0.0.2
[tests (failed: The compilation of config.0.0.2 failed at "dune build -p config -j 39 @install @runtest".)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/16922a5c4f8c89c2528a3ae77b454682ec9b814b/variant/extras,x86_32-ocaml-5.1,config.0.0.2,tests)

32-bit incompatibility:
```
# File "config/ppx.t/run.t", line 1, characters 0-0:
# /usr/bin/git --no-pager diff --no-index --color=always -u _build/.sandbox/c75ecc3b7cc9980b1195c2b5c8aca9ed/default/config/ppx.t/run.t _build/.sandbox/c75ecc3b7cc9980b1195c2b5c8aca9ed/default/config/ppx.t/run.t.corrected
# diff --git a/_build/.sandbox/c75ecc3b7cc9980b1195c2b5c8aca9ed/default/config/ppx.t/run.t b/_build/.sandbox/c75ecc3b7cc9980b1195c2b5c8aca9ed/default/config/ppx.t/run.t.corrected
# index 35895c6..2f857e2 100644
# --- a/_build/.sandbox/c75ecc3b7cc9980b1195c2b5c8aca9ed/default/config/ppx.t/run.t
# +++ b/_build/.sandbox/c75ecc3b7cc9980b1195c2b5c8aca9ed/default/config/ppx.t/run.t.corrected
# @@ -122,14 +122,27 @@
#  
#    $ dune clean
#    $ dune exec ./main.exe
# -  sys=unix
# +  File "e2e_libc.ml", line 139, characters 23-33:
# +  139 | let rand_max : c_int = 2147483647
# +                               ^^^^^^^^^^
# +  Error: Integer literal exceeds the range of representable integers of type int
# +  [1]
#  
#    $ dune clean
#    $ target_os=windows target_arch=x86 dune exec ./main.exe
# -  sys=win32
# +  File "e2e_libc.ml", line 139, characters 23-33:
# +  139 | let rand_max : c_int = 2147483647
# +                               ^^^^^^^^^^
# +  Error: Integer literal exceeds the range of representable integers of type int
# +  [1]
#  
#    $ dune clean
#    $ dune build
# +  File "e2e_libc.ml", line 139, characters 23-33:
# +  139 | let rand_max : c_int = 2147483647
# +                               ^^^^^^^^^^
# +  Error: Integer literal exceeds the range of representable integers of type int
# +  [1]
#  
#    $ dune clean
#    $ target_os=madeup dune describe pp whole_mod.ml
```

OCaml 5.2 detecting `misplaced-attribute`
```
-    $ dune clean
-    $ dune exec ./main.exe
- -  sys=unix
- +  File "e2e_libc.ml", line 652, characters 32-42:
- +  652 | let rlim_nlimits : c_int = (9 [@deprecated "not stable across os versions"])
- +                                        ^^^^^^^^^^
- +  Error (warning 53 [misplaced-attribute]): the "deprecated" attribute cannot appear in this context
```

File: packages/config/config.0.0.2/opam [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+homepage: "https://github.com/ocaml-sys/config.ml"
+bug-reports: "https://github.com/ocaml-sys/config.ml/issues"
+depends: [
+  "ocaml" {>= "5.1"}
+  "ppxlib" {>= "0.31.0"}
+  "sedlex" {>= "3.2"}
+  "spices" {>= "0.0.2"}
+  "dune" {>= "3.11"}
+  "odoc" {with-doc}
+]
```
Comment (haochenx) [inline packages/config/config.0.0.2/opam:20]: this should prevent installing with 32-bit compiler
```suggestion
]
conflicts: [ "ocaml-option-32bit" ]
```

File: packages/config/config.0.0.2/opam [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+maintainer: ["Leandro Ostera <leandro@abstractmachines.dev>"]
+authors: ["Leandro Ostera <leandro@abstractmachines.dev>"]
+license: "MIT"
+tags: [
+  "conditional" "compilation" "target_os" "target_arch" "arch" "os" "system"
+]
+homepage: "https://github.com/ocaml-sys/config.ml"
+bug-reports: "https://github.com/ocaml-sys/config.ml/issues"
+depends: [
+  "ocaml" {>= "5.1"}
```
Comment (haochenx) [inline packages/config/config.0.0.2/opam:14]: this should prevent installing with OCaml 5.2.
```suggestion
  "ocaml" {>= "5.1" & < "5.2"}
```

**Review by haochenx — COMMENTED** [review]
please also consider fixing the upstream

**Comment by leostera** [comment]
Hi @haochenx! Thanks for the review. We just dropped a test that shouldn't have been there and was failing across architectures. Keeping an eye on the CI now :)

**Review by haochenx — APPROVED** [review]
Thanks! Now every thing looks good to me.

**Comment by mseri** [comment]
Thanks

**Pull request merged**